src/strife/m_saves.c

```diff
@@ -293,8 +293,11 @@ void M_ReadMisObj(void)
 
     if((f = fopen(srcpath, "rb")))
     {
-        fread(mission_objective, 1, OBJECTIVE_LEN, f);
+        int retval = fread(mission_objective, 1, OBJECTIVE_LEN, f);
         fclose(f);
+        if (retval == 0) {
```

Comment:
`fread()` doesn't work how you think it does: what is returned is the number of items that were successfully read. It's an error if retval < OBJECTIVE_LEN, so you need to do this:

    if (retval < OBJECTIVE_LEN)
    {

Reply:
I wasn't sure if partial read here is ok or not.

Reply:
No, partial read should be considered failure.